### ISSUE TITLE
False positive 953101 when webpage contains text "file size is"

### ISSUE THREAD

**Issue by NullIsNot0** [coreruleset/coreruleset#4359]
### Description

My website has a file upload page with description text "Maximum file size is 500MB". When user opens the page, it gets blocked by CRS rule 953101 because text "file size is" is found there. I think this is too generic and too common text to be blocked by CRS.
I have written an exception rule for my upload page to work, but wanted to submit an issue so this is fixed globally.
This is something similar to #4267.

### How to reproduce the misbehavior (-> curl call)

Open the page which contains text "file size is".

### Logs

`[Mon Nov 24 07:18:21.674093 2025] [security2:error] [pid 573189:tid 573275] [client 10.20.30.40:55974] ModSecurity: Warning. Pattern match "(?i)Empty string|F(?:ile size is|reeing memory)|Header (?:name )?\\"|Invalid date|No active class|(?:Out of memor|cannot be empt)y|Pa(?:ir level|ssword is too long)|Re(?:ading file|starting!)|S(?:ession is not active|tatic function\\\\b)|T(?:elling\\\\.\\\\.\\ ..." at RESPONSE_BODY. [file "/coreruleset/rules/RESPONSE-953-DATA-LEAKAGES-PHP.conf"] [line "138"] [id "953101"] [msg "PHP Information Leakage"] [data "Matched Data: file size is found within RESPONSE_BODY"] [severity "ERROR"] [ver "OWASP_CRS/4.20.0"] [tag "application-multi"] [tag "language-php"] [tag "platform-multi"] [tag "attack-disclosure"] [tag "paranoia-level/2"] [tag "OWASP_CRS"] [tag "OWASP_CRS/DATA-LEAKAGES-PHP"] [tag "capec/1000/118/116"] [hostname "test.example.com"] [uri "/en/data-upload"] [unique_id "aSQGvbmIBMiJZfwrAu-SaAAAAPM"], referer: https://test.example.com/en/data-management`

### Your Environment

* CRS version (e.g., v3.3.4): 4.20.0
* Paranoia level setting (e.g. PL1) : PL2
* ModSecurity version (e.g., 2.9.6): 2.9.12
* Web Server and version or cloud provider / CDN (e.g., Apache httpd 2.4.54): 2.4.65
* Operating System and version: Ubuntu 25.04

### Confirmation

[x] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by michelamarie** [comment]
Hi, there. Thanks for reporting this. I had a quick internal discussion about your report with one of the project leaders, and the preferred resolution is to make the regular expression for that error string in question tighter -- if it is possible to do so in order to avoid false positives triggered by a match of the string, and also without enabling a bypass.

To explore that possibility myself, and to see exactly what the error is when it actually happens, I tried to trigger the "file size" error in some test PHP environments under my care, and I was unable to spot it, nor could a specific reference to it in the PHP documentation be found. I will continue looking for that, but in the meantime, I suggest another way to deal with this false positive that gives you better coverage than excluding the entire rule.

Basically, after you exclude the rule as you have done, add a custom rule that is identical, apart from the '(?:ile size is' bit of the expression. Here is an example:

```
SecRule RESPONSE_BODY "@rx (?i)Empty string|Freeing memory)|Header (?:name )?\"|Invalid date|No active class|(?:Out of memor|cannot be empt)y|Pa(?:ir level|ssword is too long)|Re(?:ading file|starting!)|S(?:ession is not active|tatic function\b)|T(?:elling\.\.\.|he function\b)|(?:Unknown reas|invalid opti)on|e(?:mpty password|rror reading)" \
    "id:1000,\
    phase:4,\
    block,\
    capture,\
    t:none,\
    msg:'PHP Information Leakage',\
    logdata:'Matched Data: %{TX.0} found within %{MATCHED_VAR_NAME}',\
    tag:'application-multi',\
    tag:'language-php',\
    tag:'platform-multi',\
    tag:'attack-disclosure',\
    tag:'CUSTOM_RULE/DATA-LEAKAGES-PHP',\
    severity:'ERROR',\
    setvar:'tx.outbound_anomaly_score_pl2=+%{tx.error_anomaly_score}'"
```

When I have more to report, I'll reply to this thread again.  :)

Best,
Michela

**Comment by NullIsNot0** [comment]
Thank you for suggestion @michelamarie. I'll try your suggested solution until this issue is resolved.

**Comment by RiddhiPachehara** [comment]
Thanks for the detailed report and reproduction steps.

I can confirm that the phrase “file size is” is very common in user-facing upload pages and does not necessarily indicate PHP information leakage by itself. In this context, triggering rule 953101 on static instructional text (e.g. upload size limits) seems likely to cause false positives at PL2.

I also noticed that the regex currently groups “File size is” together with other stronger PHP error indicators, which may make the match too generic when applied to RESPONSE_BODY.

Would it make sense to either:
- Narrow this specific sub-pattern with additional context, or
- Exclude matches where the phrase appears in static text without surrounding PHP error semantics?

I’m happy to help investigate similar cases or test potential refinements if guidance is needed.

**Comment by EsadCetiner** [comment]
I have a suspicion that some of the entries in `php-errors.data` and friends were added by accident. I'll assign myself to this so I can dig a bit deeper into the issue and try to prevent this issue from coming up again.
